src/main/java/org/jabref/gui/entryeditor/fileannotationtab/FileAnnotationViewModel.java

```diff
@@ -31,7 +31,11 @@ private void setupContentProperties(FileAnnotation annotation) {
             this.content.set(annotation.getLinkedFileAnnotation().getContent());
             String annotationContent = annotation.getContent();
             String illegibleTextMessage = Localization.lang("The marked area does not contain any legible text!");
-            this.marking.set(annotationContent.isEmpty() ? illegibleTextMessage : annotationContent);
+            String markingContent = (annotationContent.isEmpty() ? illegibleTextMessage : annotationContent);
+            // remove newlines && hyphens before linebreaks
+            markingContent = new RemoveHyphenatedNewlinesFormatter().format(markingContent);
```

Comment:
It is good that you try to reuse existing logic, i.e. the existing formatters. However, these are also used to generate the bibtex key (and file name). For these use cases, it is still desired to remove all line breaks. Thus, I would prefer if you define the modified regex pattern to remove the linebreaks here in this class and leave the RemoveNewlinesFormatter untouched (except changing the pattern to use `\R`).

Reply:
Now I leave the two formatter untouched and only use the pattern in this scope.